From 09f6f62fc8cc4b2e3da9df34a4e63fa8dda682f3 Mon Sep 17 00:00:00 2001
From: aniruddhgutta <aniruddhgutta@disroot.org>
Date: Wed, 22 Oct 2025 09:35:45 +0530
Subject: [PATCH] fix-vla

---
 src/screenshot/screenshot.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/screenshot/screenshot.c b/src/screenshot/screenshot.c
index 48c4c00..33dc1a8 100644
--- a/src/screenshot/screenshot.c
+++ b/src/screenshot/screenshot.c
@@ -126,25 +126,31 @@ static int method_screenshot(sd_bus_message *msg, void *data,
 	}
 
 	const char uri_prefix[] = "file://";
-	char uri[strlen(path) + strlen(uri_prefix) + 1];
-	snprintf(uri, sizeof(uri), "%s%s", uri_prefix, path);
+	size_t uri_len = strlen(path) + strlen(uri_prefix) + 1;
+	char *uri = malloc(uri_len);
+	if (!uri) return -ENOMEM;
+	snprintf(uri, uri_len, "%s%s", uri_prefix, path);
 
 	sd_bus_message *reply = NULL;
 	ret = sd_bus_message_new_method_return(msg, &reply);
 	if (ret < 0) {
+		free(uri);
 		return ret;
 	}
 
 	ret = sd_bus_message_append(reply, "ua{sv}", PORTAL_RESPONSE_SUCCESS, 1, "uri", "s", uri);
 	if (ret < 0) {
+		free(uri);
 		return ret;
 	}
 
 	ret = sd_bus_send(NULL, reply, NULL);
 	if (ret < 0) {
+		free(uri);
 		return ret;
 	}
 
+	free(uri);
 	sd_bus_message_unref(reply);
 	return 0;
 }
-- 
2.51.0

